### PR TITLE
KAFKA-17756: Added dynamic share group configs share.heartbeat.interval.ms and share.session.timeout.ms

### PR DESCRIPTION
**Pull request by chirag-wadhwa5**
This PR adds the 2 configs `share.heartbeat.interval.ms` and `share.session.timeout.ms` in GroupConfig. These can be dynamically set for a share group without restarting the server

### CODE REVIEW

File: core/src/test/java/kafka/server/share/SharePartitionManagerTest.java [hunk 1]
```diff
@@ -2368,7 +2368,7 @@ public SharePartitionManager build() {
                     persister,
                     delayedShareFetchPurgatory,
                     delayedActionsQueue,
-                    new GroupConfigManager(new HashMap<>()),
+                    mock(GroupConfigManager.class),
```
Comment (apoorvmittal10) [inline core/src/test/java/kafka/server/share/SharePartitionManagerTest.java:2371]: Hmmm, though it's fine to mock it but why we did if we are not changing any handling in SharePartitionManagerTests?

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupConfig.java [hunk 2]
```diff
@@ -63,6 +71,18 @@ public class GroupConfig extends AbstractConfig {
             atLeast(1),
             MEDIUM,
             GroupCoordinatorConfig.CONSUMER_GROUP_HEARTBEAT_INTERVAL_MS_DOC)
+        .define(SHARE_SESSION_TIMEOUT_MS_CONFIG,
+            INT,
+            GroupCoordinatorConfig.SHARE_GROUP_SESSION_TIMEOUT_MS_DEFAULT,
+            atLeast(1),
```
Comment (apoorvmittal10) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupConfig.java:77]: KIP defines `share.min.session.timeout.ms ` as 45 secs but we do not have any range defined for `share.session.timeout.ms`. Though I understand that in other places as well (`heartbeat.interval.ms`) there is no valid range defined but just writing it down to get the clarification.

**Review by apoorvmittal10 — COMMENTED** [review]
LGTM! Just minor comments for clarification.

File: core/src/test/java/kafka/server/share/SharePartitionManagerTest.java (hunk 1 above)
Comment (chirag-wadhwa5) [reply core/src/test/java/kafka/server/share/SharePartitionManagerTest.java:2371]: Hi, thanks for the review. This change should have gone in the previous PR. I got a comment to change the instance of GroupConfigManager to a mock, but I missed that change in SharePartitionManager. I can create a separate PR for this as well if you like.

Comment (apoorvmittal10) [reply core/src/test/java/kafka/server/share/SharePartitionManagerTest.java:2371]: It's fine to keep in this PR, thanks for the context.

**Review by apoorvmittal10 — APPROVED** [review]
LGTM.

**Comment by AndrewJSchofield** [comment]
@chirag-wadhwa5 There's a relevant test failure:
```
FAILED ❌ KafkaApisTest > testDescribeConfigsConsumerGroup()
```

**Review by AndrewJSchofield — APPROVED** [review]

**Review by omkreddy — APPROVED** [review]
@chirag-wadhwa5 Thanks for the PR. LGTM

**Pull request merged**